### PR TITLE
Add experimental mergify configuration

### PR DESCRIPTION
**Pull request by AlekSi**
# Description

See https://mergify.com and https://docs.mergify.com/configuration/

That configuration should remove the need to constantly click the "Update branch" button.

It probably will be replaced by a GitHub-native merge queue (https://github.com/features/merge-queue/signup) when we get access to it.

## Readiness checklist

* [ ] I added tests for new functionality or bugfixes.
* [ ] I ran `task all`, and it passed.
* [ ] I added/updated comments for both exported and unexported top-level declarations (functions, types, etc).
* [ ] I checked comments rendering with `task godocs`.
* [x] I ensured that the title is good enough for the changelog.
* [x] (for maintainers only) I set Reviewers ([`@FerretDB/core`](https://github.com/orgs/FerretDB/teams/core)), Assignee, Labels, Project and project's Sprint fields.
* [x] I marked all done items in this checklist.

### CODE REVIEW

File: .github/mergify.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+---
+# see https://docs.mergify.com/configuration/
+pull_request_rules:
+  - name: "Update out-of-date ready PRs"
+    conditions:
+      - "-closed"
+      - "-conflict"
```
Comment (rumyantseva) [inline .github/mergify.yml:7]: I looked at https://docs.mergify.com/conditions/#conditions, and from docs, I'd read this condition as "and the pull request has a conflict with the base branch". Did I misunderstand it?

**Review by rumyantseva — COMMENTED** [review]
The configuration looks strange to me - are the conditions "or" conditions or "and" conditions? It's not straightforward from the docs, but if you checked it, I'm ok with the configuration.

File: .github/mergify.yml [hunk 2]
```diff
@@ -0,0 +1,16 @@
+---
+# see https://docs.mergify.com/configuration/
+pull_request_rules:
+  - name: "Update out-of-date ready PRs"
+    conditions:
```
Comment (rumyantseva) [inline .github/mergify.yml:5]: There is `-or` operator in the docs (https://docs.mergify.com/conditions/#combining-conditions-with-operators), I'd suggest using it to be explicit.

File: .github/mergify.yml (hunk 1 above)
Comment (AlekSi) [reply .github/mergify.yml:7]: From https://docs.mergify.com/conditions/#grammar

> The optional `-` prefix is equivalent to the not operator.

Comment (rumyantseva) [reply .github/mergify.yml:7]: Ah! I actually misread this and didn't notice `-`. And I guess by default, the conditions use `and`?

File: .github/mergify.yml (hunk 2 above)
Comment (rumyantseva) [reply .github/mergify.yml:5]: Or in our case, it's rather `and`...

File: .github/mergify.yml (hunk 1 above)
Comment (AlekSi) [reply .github/mergify.yml:7]: Correct

**Comment by AlekSi** [comment]
@rumyantseva PTAL

**Review by rumyantseva — APPROVED** [review]
I'd prefer a longer syntax (`not` and `and` being mentioned explicitly) to improve readability and not confuse people, but it's not a strong opinion, so I'm ready to approve it.

**Comment by AlekSi** [comment]
All their examples use implicit `and`: https://docs.mergify.com/examples/ Just like MongoDB with filters without `$and` :)

**Review by w84thesun — APPROVED** [review]

**Review by chilagrow — APPROVED** [review]
Let's try it out!

**Pull request merged**